### PR TITLE
"No inline style" option

### PR DESCRIPTION
**Pull request by Jirinrin**
This PR has the purpose of giving the option to not insert the package's styles inline. Specifically due to the project I'm working on having this (recommended) CORS header enabled on production:
image

So 2 changes to give this option:
- allow passing an 'options' object in the constructor in which one can specify `noInlineStyle` in order to not load the floating focus styles from JS
- make a compiled CSS file appear in the NPM package, so that one can integrate it in their own bundler and/or import into a SCSS file of their own

Oh by the way, this project has no `.nvmrc` so this might be good to straighten out eventually; I installed the `mini-css-extract-plugin` package on node v10.15.3 because on node 12 the webpack stuff didn't even work :)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export default class FloatingFocus {
-	constructor(container: Element = document.body) {}
+	constructor(
+		container: Element = document.body,
+		opts: Partial<{ noInlineStyle: boolean }> = {},
+	) {}
```
Comment (Jirinrin) [inline index.d.ts:5]: I'd actually like this to just have just 1 argument being the options, with `container` being another field on that... but then it's a breaking change which feels meh ;_;

File: webpack.common.js [hunk 2]
```diff
@@ -34,6 +35,18 @@ module.exports = {
...
 			},
+			{
+				test: /\.(css|scss)$/,
+				use: [
+					MiniCssExtractPlugin.loader,
+					'css-loader',
+					'postcss-loader',
+					'sass-loader'
+				]
+			},
```
Comment (Jirinrin) [inline webpack.common.js:46]: Hrmm when I made this I was sure I tested it thoroughly by importing the code into the project where I want to use it... but now (using example.html), the CSS can only get included in the bundle if I don't include this extra rule...
Is there a good way I don't know of to sort of "in parallel" extract the css to a separate file and also include it in the bundle?

**Review by guidobouman — CHANGES_REQUESTED** [review]
Change the architecture to create a setup that's more flexible, and will prevent duplicate inclusion of style in your bundles.

File: index.d.ts (hunk 1 above)
Comment (Jirinrin) [reply index.d.ts:5]: Ah that works for me (:

**Comment by Jirinrin** [comment]
In the end I actually like making a separate unstyled build quite a bit, because it just seems a slight bit cleaner to not have those 900 characters of CSS not included twice if you don't want them (though I guess 900 characters is quite negligible in terms of bundle size)

File: webpack.prod.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 const merge = require('webpack-merge');
-const common = require('./webpack.common.js');
-const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const styled = require('./webpack.styled.js');
+const unstyled = require('./webpack.unstyled.js');
 
-module.exports = merge(common, {
-	mode: 'production',
-	plugins: [
-		new CleanWebpackPlugin()
```
Comment (Jirinrin) [inline webpack.prod.js:9]: The CleanWebpackPlugin needed to go, because with it, it would only keep the files of one of the two builds.

Comment (guidobouman) [reply webpack.prod.js:9]: How is it possible that it does work with all NLO verticals? 🤔 

https://github.com/nederlandseloterij/NLO.Lotto/blob/develop/src/NLO.Lotto/webpack.prod.js

File: package.json [hunk 4]
```diff
@@ -35,15 +36,16 @@
...
 		"core-js": "^3.2.1",
 		"css-loader": "^3.2.0",
 		"eslint": "^6.3.0",
 		"eslint-loader": "^3.0.0",
 		"jest": "^24.9.0",
 		"jest-cli": "^24.9.0",
 		"jest-transform-stub": "^2.0.0",
+		"mini-css-extract-plugin": "^0.9.0",
 		"node-sass": "^4.13.0",
+		"parallel-webpack": "^2.6.0",
```
Comment (guidobouman) [inline package.json:48]: Is that the cause of the double clean error?

File: webpack.common.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 /* global __dirname */
 'use strict';
 const path = require('path');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
```
Comment (guidobouman) [inline webpack.common.js:4]: Is this still required when it's also in the unstyled webpack file?

**Review by guidobouman — CHANGES_REQUESTED** [review]
Nice work!

File: package.json (hunk 4 above)
Comment (Jirinrin) [reply package.json:48]: No it doesn't matter if the two builds are done consecutively or concurrently -- there will be two writes to the directory regardless

File: webpack.common.js (hunk 5 above)
Comment (Jirinrin) [reply webpack.common.js]: Good catch, haha... yes I should've removed that

**Comment by guidobouman** [comment]
@Jirinrin See https://github.com/Q42/floating-focus-a11y/pull/22#discussion_r513774841

File: webpack.prod.js (hunk 3 above)
Comment (Jirinrin) [reply webpack.prod.js:9]: Haha I don't have access to this organisation anymore °~°

**Comment by Jirinrin** [comment]
Good find, interestingly I wasn't able to find that option...
I've now brought clean-webpack-plugin back, with the exclude patterns suggestion -- I also wouldn't have minded separate build folders for `styled` and `unstyled` (and then the js file in `unstyled` could be `index.js` so the import statement would remain the same), but it feels a little overkill-ish to me for just these three files?

File: package.json [hunk 6]
```diff
@@ -20,10 +20,11 @@
 	"bugs": {
 		"url": "https://github.com/Q42/FloatingFocus/issues"
 	},
-	"main": "dist/floating-focus.js",
+	"main": "dist/styled/floating-focus.js",
```
Comment (guidobouman) [inline package.json:23]: I kinda like this foldered output. (And right ow it's broken without the output. 😏 )

**Review by guidobouman — CHANGES_REQUESTED** [review]
Sorry for the unending request changes on this one... 😅

File: package.json (hunk 6 above)
Comment (Jirinrin) [reply package.json:23]: pff did I forget removing this one? I'm also becoming less sharp haha

Comment (Jirinrin) [reply package.json:23]: Okay but I'll take another shot at doing it in the foldered way (:

**Review by guidobouman — APPROVED** [review]

**Comment by guidobouman** [comment]
I did some tweaks. Do you agree? :D

File: README.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
-# floating-focus-a11y [![Build Status](https://travis-ci.com/Q42/floating-focus-a11y.svg?token=zMS2E6VVY9WYyfREUSjH&branch=master)](https://travis-ci.com/Q42/floating-focus-a11y) [![npm version](https://badge.fury.io/js/%40q42%2Ffloating-focus-a11y.svg)](https://badge.fury.io/js/%40q42%2Ffloating-focus-a11y)
-A clear, beautiful and easy to implement focus-state solution.
+# Accessible Floating Focus [![Build Status](https://travis-ci.com/Q42/floating-focus-a11y.svg?token=zMS2E6VVY9WYyfREUSjH&branch=master)](https://travis-ci.com/Q42/floating-focus-a11y) [![npm version](https://badge.fury.io/js/%40q42%2Ffloating-focus-a11y.svg)](https://badge.fury.io/js/%40q42%2Ffloating-focus-a11y)
+A clear, beautiful and easy to implement focus-state solution that improves accessibility in an aestetically pleasing way.
```
Comment (Jirinrin) [inline README.md:2]: ``
```suggestion
A clear, beautiful and easy to implement focus-state solution that improves accessibility in an aesthetically pleasing way.
```

**Review by Jirinrin — COMMENTED** [review]
Yes, I _do_ agree :3
(except for that one typo)

**Pull request merged**